src/libs/test-libs/utillib/convert/testing.cpp

```diff
@@ -168,6 +168,36 @@ TEST_F(CThisTest, TestConverts_3) {
     return true;
 }}
 
+//------------------------------------------------------------------------
+TEST_F(CThisTest, TestConverts_4) {
+        cout << "Running " << testName << "\n";
+
+        ASSERT_EQ("double_2_Str1", double_2_Str(14.15019, 5), "14.15019")
+        ASSERT_EQ("double_2_Str2", double_2_Str(14.15019, 3), "14.150")
+        ASSERT_EQ("double_2_Str3", double_2_Str(10586816.10315913591, 10), "10586816.1031591352") // should be 10586816.1031591359???
+        ASSERT_EQ("double_2_Str4", double_2_Str(10586816.10315913591, 2), "10586816.10")
+        ASSERT_EQ("double_2_Str5", double_2_Str(1050, 3), "1050.000")
+        ASSERT_EQ("double_2_Str6", double_2_Str(0.150100005, 9), "0.150100005")
+        ASSERT_EQ("double_2_Str7", double_2_Str(-14.15019, 5), "-14.15019")
+        ASSERT_EQ("double_2_Str8", double_2_Str(-14.15019, 3), "-14.150")
+        ASSERT_EQ("double_2_Str9", double_2_Str(-10586816.10315913591, 10), "-10586816.1031591352") // should be -10586816.1031591359???
+        ASSERT_EQ("double_2_Str10", double_2_Str(-10586816.10315913591, 2), "-10586816.10")
+        ASSERT_EQ("double_2_Str11", double_2_Str(-1050, 3), "-1050.000")
+        ASSERT_EQ("double_2_Str12", double_2_Str(-0.150100005, 9), "-0.150100005")
+
+        //ASSERT_EQ("double_2_Str13", double_2_Str(10586816.10315913591, 11), "10586816.10315913591")
+
+        // Significant figures
+        ASSERT_EQ("double_2_Str14", double_2_Str(1.100000000005), "1.1")
+        ASSERT_EQ("double_2_Str15", double_2_Str(0.), "0")
+        ASSERT_EQ("double_2_Str15", double_2_Str(0.000000000), "0")
+
+        ASSERT_EQ("addr_2_Str1", addr_2_Str("0x281055afc982d96fab65b3a49cac8b878184cb16"), "0x281055afc982d96fab65b3a49cac8b878184cb16")
+
```

Comment:
Let's add two more test cases:
    double_2_Str(-1e-10, 4) == "-0.0001"
    double_2_Str(128,-1) == "120"

Reply:
@tjayrush the test cases you mention about should be `ASSERT_NOT_EQ`. Is that what you had in mind?

Reply:
I made a mistake.  -1e4,5 should be -0.0001 And I just wanted to see what would happen with a negative input for width.